tests/mqtt/test_adapter.py

```
@@ -0,0 +1,735 @@
+
+    def test_disconnecting_client_twice_has_no_effect(self):
+        self.adapter.connect()
+        self.adapter.disconnect()
+        self.adapter.disconnect()
+        self.assertFalse(self.adapter.is_connected)
+
+    @patch("paho.mqtt.client.Client.disconnect")
+    def test_error_is_logged_when_non_ok_return_code_is_returned_from_mqtt_client(self, mock: Mock):
+        mock.return_value = MQTT_ERR_SUCCESS + 1
+        self.adapter.connect()
+        with self.assertLogs(logger=_logger._logger, level=logging.ERROR) as cm:
```

Ruff

Local variable `cm` is assigned to but never used

Remove assignment to unused variable `cm`

(F841)